score/score_test.go

```diff
@@ -0,0 +1,66 @@
+package score
+
+import "testing"
+
+var theSecret = "my secret code"
+
+func init() {
+	GlobalSecret = theSecret
+}
+
+var nonJSONLog = []string{
+	"heere is some output",
+	"some other output",
+	"line contains " + theSecret,
+	theSecret + " should not be revealed",
+}
+
+func TestParseNonJSONStrings(t *testing.T) {
+	for _, s := range nonJSONLog {
+		sc, err := Parse(s, GlobalSecret)
+		if err == nil {
+			t.Errorf("Expected '%v', got '<nil>'", ErrScoreNotFound.Error())
+		}
+		if sc != nil {
+			t.Errorf("Got unexpected score object '%v', wanted '<nil>'", sc)
+		}
+	}
+}
+
+var jsonLog = []struct {
+	in  string
+	out *Score
+	err error
+}{
+	{`{"Secret":"` + theSecret + `","TestName":"init","Score":0,"MaxScore":10,"Weight":10}`,
+		NewScore(10, 10),
+		nil,
+	},
+	{`{"Secret":"the wrong secret","TestName":"init","Score":0,"MaxScore":10,"Weight":10}`,
+		nil,
+		ErrScoreNotFound,
+	},
+}
+
+// Equal returns true if sc equals other. Ignores the Secret field.
```

Comment:
Its possible, but right now I don't see any applications for comparing Score objects elsewhere in the code, but if you think it could be useful, I don't mind moving it. As it is now, the `Equal()` function is only compiled only when we test, and otherwise it is not compiled (available.) But again, feel free to add it.